activerecord/test/cases/relation_test.rb

```diff
@@ -293,6 +293,15 @@ def test_relation_merging_with_joins_as_join_dependency_pick_proper_parent
       assert_equal 3, relation.where(id: post.id).pluck(:id).size
     end
 
+    def test_merge_preserves_duplicate_columns
+      quoted_posts_id = Regexp.escape(quote_table_name("posts.id"))
+      quoted_posts = Regexp.escape(quote_table_name("posts"))
+      posts = Post.select(:id)
+      assert_queries_match(/SELECT #{quoted_posts_id}, #{quoted_posts_id} FROM #{quoted_posts}/i) do
+        posts.merge(posts).to_a
```

Comment:
I'm ok to allow explicit `select(:id, :id)` duplication, but it is not the expected behavior that values that users has not touched are duplicated each time `merge` is called. Fixed by 31d3caece6be5511dbfb5269f926dcf1648b59ac.